### PR TITLE
feat(cli): Extend existing list command to output only a list of file names

### LINKED ISSUE #6374: Extend existing `list` command to output only a list of file names
### Clear and concise description of the problem

https://github.com/vitest-dev/vitest/issues/2901 lead to the introduction of the [`vitest list`](https://vitest.dev/guide/cli.html#vitest-list) command, but unfortunately this is pretty slow and requires some processing if you're only interested in a list of _files_ that will run with the given filters & flags (i.e. `--changed`), since it parses all test files and returns a list containing every individual test case. What I ideally would want is a command that returns which files are going to be executed given the provided filters & flags without drilling down to individual test cases, requiring only a lookup and eliminating the need for a parse, thus being a lot quicker.

My main use-case for this is dynamic sharding, where I dynamically determine the amount of shards and agents I want to spin up based on a predetermined number `TESTS_PER_SHARD` (i.e. `100`), which serves to keep the test performance stable, irregardless of number of tests. Jest for example can achieve this with [`--listTests [--json]`](https://jestjs.io/docs/cli#--listtests), which runs in under a second for ~1k test files and leads to the following output:

**`jest --listTests`**

```
/path/to/testA.spec.tsx
/path/to/testB.spec.tsx
```
**`jest --listTests --json`**

```
["/path/to/testA.spec.tsx", "/path/to/testB.spec.tsx"]
```

For comparison, running `vitest list --json` on an M1 MBP for the same ~1k files takes 3.5 minutes.

### Suggested solution

Since there is already a dedicated `list` command I suppose it would make sense to provide a new optional flag, i.e. `--filesOnly`.

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by Ma-hawaj**
- **feat(cli): add the files only option to the list command (unfuncitonal)**
- **feat(cli): add implementation for listing files only**
- **feat(cli): add testing for list --filesOnly**
- **feat(cli): add documentation for list --filesOnly**

### Description

Extend existing list command to output only a list of file names

closes #6374

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or 
introduces breaking changes without a discussion, PR might be closed.
- [ ] Ideally, include a test that fails without this PR but passes with it.
- [x] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [ ] Run the tests with `pnpm test:ci`.

### Documentation
- [x] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

**Review by sheremet-va — CHANGES_REQUESTED** [review]
This implementation seems a bit pointless because you can do the same yourself on `vitest list` result.

What we need is to _not collect_ tests at all to make this very fast. This line should just not be called:

https://github.com/vitest-dev/vitest/blob/95f0203f27f5659f5758638edc4d1d90283801ac/packages/vitest/src/node/core.ts#L369

**Comment by Ma-hawaj** [comment]
OK, 
so will adding the following
```
async collectFiles(filters?: string[]) {
    const files = await this.filterTestsBySource(
      await this.globTestFiles(filters),
    )
}
```
in the vitest class and checking the option in cac.ts:collect() and calling it will do the job.

File: packages/vitest/src/node/cli/cli-api.ts [hunk 1]
```diff
@@ -184,6 +189,51 @@ export function processCollected(ctx: Vitest, files: File[], options: CliOptions
...
+  })
+}
+
+function formatFilesAsString(files: WorkspaceSpec[]) {
+  return files.map((file) => {
+    let name = file.moduleId.substring(String(file[0].runner.root).length + 1)
+    if (file.project.name) {
+      name = `[${file.project.name}] ${name}`
+    }
+    console.log(name)
```
Comment (sheremet-va) [inline packages/vitest/src/node/cli/cli-api.ts:232]: Why does the `format` function has `console.log`?

**Review by sheremet-va — CHANGES_REQUESTED** [review]

File: packages/vitest/src/node/cli/cli-api.ts (hunk 1 above)
Comment (Ma-hawaj) [reply packages/vitest/src/node/cli/cli-api.ts:232]: It also prints the output to the user at the same time

File: packages/vitest/src/node/cli/cli-api.ts [hunk 2]
```diff
@@ -184,6 +189,50 @@ export function processCollected(ctx: Vitest, files: File[], options: CliOptions
   return formatCollectedAsString(files).forEach(test => console.log(test))
 }
 
+export function outputFileList(files: WorkspaceSpec[], options: CliOptions) {
+  // const paths = files.map((file) => file.moduleId)
```
Comment (sheremet-va) [inline packages/vitest/src/node/cli/cli-api.ts:193]:
```suggestion
```

File: packages/vitest/src/node/cli/cli-api.ts [hunk 3]
```diff
@@ -184,6 +189,48 @@ export function processCollected(ctx: Vitest, files: File[], options: CliOptions
...
+    if (file.project.name) {
+      result.projectName = file.project.name
+    }
+    return result
+  })
+}
+
+function formatFilesAsString(files: WorkspaceSpec[]) {
+  return files.map((file) => {
+    let name = file.moduleId.substring(String(file[0].runner.root).length + 1)
```
Comment (sheremet-va) [inline packages/vitest/src/node/cli/cli-api.ts:226]:
```suggestion
    let name = resolve(options.root || process.cwd(), file.moduleId)
```

**Review by sheremet-va — CHANGES_REQUESTED** [review]

File: packages/vitest/src/node/cli/cli-api.ts [hunk 4]
```diff
@@ -184,6 +189,48 @@ export function processCollected(ctx: Vitest, files: File[], options: CliOptions
...
+    }
+
+    if (file.project.name) {
+      result.projectName = file.project.name
+    }
+    return result
+  })
+}
+
+function formatFilesAsString(files: WorkspaceSpec[]) {
```
Comment (Ma-hawaj) [inline packages/vitest/src/node/cli/cli-api.ts:224]:
```suggestion
function formatFilesAsString(files: WorkspaceSpec[], options: CliOptions) {
```

File: packages/vitest/src/node/cli/cli-api.ts [hunk 5]
```diff
@@ -184,6 +189,48 @@ export function processCollected(ctx: Vitest, files: File[], options: CliOptions
   return formatCollectedAsString(files).forEach(test => console.log(test))
 }
 
+export function outputFileList(files: WorkspaceSpec[], options: CliOptions) {
+  if (typeof options.json !== 'undefined') {
+    return outputJsonFileList(files, options)
+  }
+
+  return formatFilesAsString(files).map(file => console.log(file))
```
Comment (Ma-hawaj) [inline packages/vitest/src/node/cli/cli-api.ts:197]:
```suggestion
  return formatFilesAsString(files, options).map(file => console.log(file))
```

File: packages/vitest/src/node/cli/cli-api.ts (hunk 3 above)
Comment (Ma-hawaj) [reply packages/vitest/src/node/cli/cli-api.ts]: > 

Seems this didn't work, it prints the whole path now.

File: packages/vitest/src/node/cli/cli-api.ts [hunk 6]
```diff
@@ -184,6 +189,48 @@ export function processCollected(ctx: Vitest, files: File[], options: CliOptions
...
+
+  return formatFilesAsString(files, options).map(file => console.log(file))
+}
+
+function outputJsonFileList(files: WorkspaceSpec[], options: CliOptions) {
+  if (typeof options.json === 'boolean') {
+    return console.log(JSON.stringify(formatFilesAsJSON(files), null, 2))
+  }
+  if (typeof options.json === 'string') {
+    const jsonPath = resolve(options.root || process.cwd(), options.json)
```
Comment (sheremet-va) [inline packages/vitest/src/node/cli/cli-api.ts:205]:
```suggestion
    const jsonPath = relative(options.root || process.cwd(), options.json)
```

should be imported from `pathe`, can't add the comment there unfortunetly

Comment (Ma-hawaj) [reply packages/vitest/src/node/cli/cli-api.ts:205]: did it in 99afa71

**Comment by sheremet-va** [comment]
Thanks!

**Pull request merged**